code/datums/quirks/_quirk.dm

```diff
@@ -280,3 +281,20 @@
 	for(var/datum/quirk/quirk as anything in quirks)
 		quirk.remove_from_current_holder(quirk_transfer = TRUE)
 		quirk.add_to_holder(to_mob, quirk_transfer = TRUE, client_source = to_pass)
+
+
+/datum/quirk/add_to_holder(mob/living/new_holder, quirk_transfer, client/client_source)
+	if(!can_add(new_holder))
+		CRASH("Attempted to add quirk to holder that can't have it.")
+	. = ..()
```

Comment:
this should not be done in the modular style, we don't segment procs like this except for extreme cases